roles/validate/files/rules/required_rules/502_policy_vrf_lite_cross_reference.py

```diff
@@ -0,0 +1,257 @@
+'''
+Validation Rules scenarios:
+1.  Switches in the [insert vrf-lite switch model key structure here] should be defined in the vxlan.topology.switches
+2.  BGP and OSPF should be defined in two different policy.
+3.  route_reflector_client for AF IPv4 and IPv6 config should not be allowed.
+4.  Check if OSPF Authentication is enabled, key must be provided
+5.  Check AREA 0 is standard
+6.  Check static routes across Policies
+7.  Check ospf network type for Loopback
+'''
+
+
+class Rule:
+    id = "502"
+    description = "Verify VRF-Lites Cross Reference Between Policies, Groups, and Switches"
+    severity = "HIGH"
+
+    @classmethod
+    def match(cls, inventory):
+        vrf_lites = []
+        topology_switches = []
+        policies = []
+        results = []
+        switch_policy = []
+        static_routes_compliance = []
+
+        # Get fabric switches
+        if inventory.get("vxlan"):
+            if inventory["vxlan"].get("topology"):
+                if inventory.get("vxlan").get("topology").get("switches"):
+                    topology_switches = inventory.get("vxlan").get("topology").get("switches")
+
+        # Get vrf-lites policies
+        if inventory.get("vxlan", None):
+            if inventory["vxlan"].get("overlay_extensions", None):
+                if inventory["vxlan"].get("overlay_extensions").get("vrf_lites", None):
+                    vrf_lites = inventory["vxlan"]["overlay_extensions"]["vrf_lites"]
+
+        for policy in vrf_lites:   # Check Global Level
+            if policy.get('name', None):
+                policies.append(policy['name'])
+
+                cls.check_global_ospf_and_bgp(results, policy)
```

Comment:
Instead of passing in results to each function you should be able to make it a class variable.

Change line 23 above to `self.results`

Reply:
not possible to to update with self.results. It's limited by iac-validate script.

Reply:
Did you try with a dunder init?

Reply:
Done. Move results variable under the class and remove result from function